From 184c28795bb98ad14bdfcef01a475d5ba11e40d5 Mon Sep 17 00:00:00 2001
From: Kristian Klausen <klausenbusk@hotmail.com>
Date: Fri, 22 Sep 2017 17:54:14 +0000
Subject: [PATCH 35/48] alsa-mixer: Prioritize hdmi-* mappings over iec958-*
 mappings

Pulseaudio tries to pick the best profile (on startup or
hotplugged), the best profile is the profile with the highest
priority which isn't unavailable.
Due to the facts that iec958 ports available status always (?)
is unknown, and that it is generally more likely that a user use
hdmi than iec958, lets prioritze hdmi over iec958.

This patch shift the analog-* mappings +5 and hdmi-* mappings +5.
---
 src/modules/alsa/mixer/profile-sets/default.conf | 80 ++++++++++++------------
 1 file changed, 40 insertions(+), 40 deletions(-)

diff --git a/src/modules/alsa/mixer/profile-sets/default.conf b/src/modules/alsa/mixer/profile-sets/default.conf
index c360e772..f5093830 100644
--- a/src/modules/alsa/mixer/profile-sets/default.conf
+++ b/src/modules/alsa/mixer/profile-sets/default.conf
@@ -106,14 +106,14 @@ device-strings = hw:%f
 channel-map = mono
 paths-output = analog-output analog-output-lineout analog-output-speaker analog-output-headphones analog-output-headphones-2 analog-output-mono
 paths-input = analog-input-front-mic analog-input-rear-mic analog-input-internal-mic analog-input-dock-mic analog-input analog-input-mic analog-input-linein analog-input-aux analog-input-video analog-input-tvtuner analog-input-fm analog-input-mic-line analog-input-headset-mic
-priority = 2
+priority = 7
 
 [Mapping analog-stereo]
 device-strings = front:%f
 channel-map = left,right
 paths-output = analog-output analog-output-lineout analog-output-speaker analog-output-headphones analog-output-headphones-2
 paths-input = analog-input-front-mic analog-input-rear-mic analog-input-internal-mic analog-input-dock-mic analog-input analog-input-mic analog-input-linein analog-input-aux analog-input-video analog-input-tvtuner analog-input-fm analog-input-mic-line analog-input-headphone-mic analog-input-headset-mic
-priority = 10
+priority = 15
 
 # If everything else fails, try to use hw:0 as a stereo device.
 [Mapping stereo-fallback]
@@ -128,35 +128,35 @@ priority = 1
 device-strings = surround21:%f
 channel-map = front-left,front-right,lfe
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 8
+priority = 13
 direction = output
 
 [Mapping analog-surround-40]
 device-strings = surround40:%f
 channel-map = front-left,front-right,rear-left,rear-right
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 7
+priority = 12
 direction = output
 
 [Mapping analog-surround-41]
 device-strings = surround41:%f
 channel-map = front-left,front-right,rear-left,rear-right,lfe
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 8
+priority = 13
 direction = output
 
 [Mapping analog-surround-50]
 device-strings = surround50:%f
 channel-map = front-left,front-right,rear-left,rear-right,front-center
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 7
+priority = 12
 direction = output
 
 [Mapping analog-surround-51]
 device-strings = surround51:%f
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 8
+priority = 13
 direction = output
 
 [Mapping analog-surround-71]
@@ -164,7 +164,7 @@ device-strings = surround71:%f
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
 description = Analog Surround 7.1
 paths-output = analog-output analog-output-lineout analog-output-speaker
-priority = 7
+priority = 12
 direction = output
 
 [Mapping iec958-stereo]
@@ -200,7 +200,7 @@ description = Digital Stereo (HDMI)
 device-strings = hdmi:%f
 paths-output = hdmi-output-0
 channel-map = left,right
-priority = 4
+priority = 9
 direction = output
 
 [Mapping hdmi-surround]
@@ -208,7 +208,7 @@ description = Digital Surround 5.1 (HDMI)
 device-strings = hdmi:%f
 paths-output = hdmi-output-0
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 3
+priority = 8
 direction = output
 
 [Mapping hdmi-surround71]
@@ -216,7 +216,7 @@ description = Digital Surround 7.1 (HDMI)
 device-strings = hdmi:%f
 paths-output = hdmi-output-0
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 3
+priority = 8
 direction = output
 
 [Mapping hdmi-dts-surround]
@@ -224,7 +224,7 @@ description = Digital Surround 5.1 (HDMI/DTS)
 device-strings = dcahdmi:%f
 paths-output = hdmi-output-0
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra1]
@@ -232,7 +232,7 @@ description = Digital Stereo (HDMI 2)
 device-strings = hdmi:%f,1
 paths-output = hdmi-output-1
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra1]
@@ -240,7 +240,7 @@ description = Digital Surround 5.1 (HDMI 2)
 device-strings = hdmi:%f,1
 paths-output = hdmi-output-1
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra1]
@@ -248,7 +248,7 @@ description = Digital Surround 7.1 (HDMI 2)
 device-strings = hdmi:%f,1
 paths-output = hdmi-output-1
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra1]
@@ -256,7 +256,7 @@ description = Digital Surround 5.1 (HDMI 2/DTS)
 device-strings = dcahdmi:%f,1
 paths-output = hdmi-output-1
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra2]
@@ -264,7 +264,7 @@ description = Digital Stereo (HDMI 3)
 device-strings = hdmi:%f,2
 paths-output = hdmi-output-2
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra2]
@@ -272,7 +272,7 @@ description = Digital Surround 5.1 (HDMI 3)
 device-strings = hdmi:%f,2
 paths-output = hdmi-output-2
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra2]
@@ -280,7 +280,7 @@ description = Digital Surround 7.1 (HDMI 3)
 device-strings = hdmi:%f,2
 paths-output = hdmi-output-2
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra2]
@@ -288,7 +288,7 @@ description = Digital Surround 5.1 (HDMI 3/DTS)
 device-strings = dcahdmi:%f,2
 paths-output = hdmi-output-2
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra3]
@@ -296,7 +296,7 @@ description = Digital Stereo (HDMI 4)
 device-strings = hdmi:%f,3
 paths-output = hdmi-output-3
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra3]
@@ -304,7 +304,7 @@ description = Digital Surround 5.1 (HDMI 4)
 device-strings = hdmi:%f,3
 paths-output = hdmi-output-3
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra3]
@@ -312,7 +312,7 @@ description = Digital Surround 7.1 (HDMI 4)
 device-strings = hdmi:%f,3
 paths-output = hdmi-output-3
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra3]
@@ -320,7 +320,7 @@ description = Digital Surround 5.1 (HDMI 4/DTS)
 device-strings = dcahdmi:%f,3
 paths-output = hdmi-output-3
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra4]
@@ -328,7 +328,7 @@ description = Digital Stereo (HDMI 5)
 device-strings = hdmi:%f,4
 paths-output = hdmi-output-4
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra4]
@@ -336,7 +336,7 @@ description = Digital Surround 5.1 (HDMI 5)
 device-strings = hdmi:%f,4
 paths-output = hdmi-output-4
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra4]
@@ -344,7 +344,7 @@ description = Digital Surround 7.1 (HDMI 5)
 device-strings = hdmi:%f,4
 paths-output = hdmi-output-4
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra4]
@@ -352,7 +352,7 @@ description = Digital Surround 5.1 (HDMI 5/DTS)
 device-strings = dcahdmi:%f,4
 paths-output = hdmi-output-4
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra5]
@@ -360,7 +360,7 @@ description = Digital Stereo (HDMI 6)
 device-strings = hdmi:%f,5
 paths-output = hdmi-output-5
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra5]
@@ -368,7 +368,7 @@ description = Digital Surround 5.1 (HDMI 6)
 device-strings = hdmi:%f,5
 paths-output = hdmi-output-5
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra5]
@@ -376,7 +376,7 @@ description = Digital Surround 7.1 (HDMI 6)
 device-strings = hdmi:%f,5
 paths-output = hdmi-output-5
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra5]
@@ -384,7 +384,7 @@ description = Digital Surround 5.1 (HDMI 6/DTS)
 device-strings = dcahdmi:%f,5
 paths-output = hdmi-output-5
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra6]
@@ -392,7 +392,7 @@ description = Digital Stereo (HDMI 7)
 device-strings = hdmi:%f,6
 paths-output = hdmi-output-6
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra6]
@@ -400,7 +400,7 @@ description = Digital Surround 5.1 (HDMI 7)
 device-strings = hdmi:%f,6
 paths-output = hdmi-output-6
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra6]
@@ -408,7 +408,7 @@ description = Digital Surround 7.1 (HDMI 7)
 device-strings = hdmi:%f,6
 paths-output = hdmi-output-6
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra6]
@@ -416,7 +416,7 @@ description = Digital Surround 5.1 (HDMI 7/DTS)
 device-strings = dcahdmi:%f,6
 paths-output = hdmi-output-6
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-stereo-extra7]
@@ -424,7 +424,7 @@ description = Digital Stereo (HDMI 8)
 device-strings = hdmi:%f,7
 paths-output = hdmi-output-7
 channel-map = left,right
-priority = 2
+priority = 7
 direction = output
 
 [Mapping hdmi-surround-extra7]
@@ -432,7 +432,7 @@ description = Digital Surround 5.1 (HDMI 8)
 device-strings = hdmi:%f,7
 paths-output = hdmi-output-7
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-surround71-extra7]
@@ -440,7 +440,7 @@ description = Digital Surround 7.1 (HDMI 8)
 device-strings = hdmi:%f,7
 paths-output = hdmi-output-7
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right
-priority = 1
+priority = 6
 direction = output
 
 [Mapping hdmi-dts-surround-extra7]
@@ -448,7 +448,7 @@ description = Digital Surround 5.1 (HDMI 8/DTS)
 device-strings = dcahdmi:%f,7
 paths-output = hdmi-output-7
 channel-map = front-left,front-right,rear-left,rear-right,front-center,lfe
-priority = 1
+priority = 6
 direction = output
 
 [Mapping multichannel-output]
-- 
2.13.6

